eng/pipelines/templates/stages/archetype-sdk-tests.yml

```diff
@@ -0,0 +1,117 @@
+  - name: AfterTestSteps
+    type: object
+    default: []
+  - name: BuildTargetingString
+    type: string
+    default: 'azure-*'
+  - name: AdditionalTestArgs
+    type: string
+    default: ''
+  - name: TestMarkArgument
+    type: string
+    default: ''
+  - name: InjectedPackages
+    type: string
+    default: ''
+  - name: BuildDocs
+    type: boolean
+    default: true
+  - name: JobName
+    type: string
+    default: 'Test'
+  - name: AllocateResourceGroup
+    type: boolean
+    default: true
+  - name: DeployArmTemplate
+    type: boolean
+    default: false
+  - name: TestTimeoutInMinutes
+    type: number
+    default: 120
+  - name: TestSamples
+    type: boolean
+    default: false
+  - name: Location
+    type: string
+    default: ''
+  - name: Matrix
+    type: object
+    default:
+      Linux_Python35:
+        Pool: azsdk-pool-mms-ubuntu-1804-general
+        OSVmImage: MMSUbuntu18.04
+        PythonVersion: '3.5'
+        CoverageArg: '--disablecov'
+      MacOs_Python37:
+        Pool: Azure Pipelines
+        OSVmImage: 'macOS-10.15'
+        PythonVersion: '3.7'
+        CoverageArg: '--disablecov'
+      Windows_Python27:
+        Pool: azsdk-pool-mms-win-2019-general
+        OSVmImage: MMS2019
+        PythonVersion: '2.7'
+        CoverageArg: '--disablecov'
+      Linux_PyPy3:
+        Pool: azsdk-pool-mms-ubuntu-1804-general
+        OSVmImage: MMSUbuntu18.04
+        PythonVersion: 'pypy3'
+        CoverageArg: '--disablecov'
+      Linux_Python39:
+        Pool: azsdk-pool-mms-ubuntu-1804-general
+        OSVmImage: MMSUbuntu18.04
+        PythonVersion: '3.9'
+        CoverageArg: ''
+  - name: CloudConfigurations
+    type: object
+    default:
+      AzureCloud:
+        SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
+
+
+stages:
+  - stage: Build
+    jobs:
+    - template: ../jobs/archetype-sdk-tests.yml
+      parameters:
+        ServiceDirectory: ${{ parameters.ServiceDirectory }}
+        EnvVars: ${{ parameters.EnvVars }}
+        MaxParallel: ${{ parameters.MaxParallel }}
+        BeforeTestSteps: ${{ parameters.BeforeTestSteps }}
+        AfterTestSteps: ${{ parameters.AfterTestSteps }}
+        AdditionalTestArgs: ${{ parameters.AdditionalTestArgs }}
+        BuildTargetingString: ${{ parameters.BuildTargetingString }}
+        TestMarkArgument: ${{ parameters.TestMarkArgument }}
+        InjectedPackages: ${{ parameters.InjectedPackages }}
+        BuildDocs: ${{ parameters.BuildDocs }}
+        JobName: ${{ parameters.JobName }}
+        AllocateResourceGroup: ${{ parameters.AllocateResourceGroup }}
+        DeployArmTemplate: ${{ parameters.DeployArmTemplate }}
+        TestTimeoutInMinutes: ${{ parameters.TestTimeoutInMinutes }}
+        TestSamples: ${{ parameters.TestSamples }}
+        Location: ${{ parameters.Location }}
+        Matrix: ${{ parameters.Matrix }}
+        CloudConfigurations: ${{ parameters.CloudConfigurations }}
+
+
+  - stage: Publish_Coverage
```

Comment:
Why make this a separate stage/job?

Reply:
I would call the stage something other than `Build`. Maybe `LiveTest` or just leave it blank

Reply:
@weshaggard the code coverage is dependent on all the Python 3.9 jobs.

Reply:
So this this is an aggregate coverage report publishing? Is there advantage to do it this way as opposed to letting each job publish the results individually? If there is an advantage to doing an aggregate then this approach should work but I'd suggest updating the name to indicate that is what is doing.

Reply:
If each job publishes individually they will overwrite previous reports. This method guarantees that all jobs have an accurate code coverage report.

Reply:
I'm not opposed to doing an aggregate report and publishing but just curious why we couldn't solve this problem by giving each job a unique name?

Reply:
@weshaggard yes, the code coverage reports are currently over-writing each other each time they are published. This ensures that one report is published with all the code coverage for the package in one place.